addons/l10n_it_edi_imp/models/account_move.py

```diff
@@ -0,0 +1,105 @@
+            ('TD09', "TD09 - Simplified debit note"),
+            ('TD24', "TD24 - Deferred invoice (Article 21, paragraph 4, third period letter to Presidential Decree 633/72)"),
+            ('TD25', "TD25 - Deferred invoice (Article 21, paragraph 4, third period, letter b (Dropshipping))"),
+            ('TD26', "TD26 - Transfer of depreciable assets and for internal transfers (ex art.36 Presidential Decree 633/72)"),
+            ('TD27', "TD27 - Self-invoice for self-consumption or for free transfers without recourse"),
+          ],
+        compute='_compute_l10n_it_document_type',
+        store=True,
+        readonly=False,
+    )
+
+    l10n_it_document_type_purchase = fields.Selection(
+        selection=[
+            ('TD16', "TD16 - Internal reverse charge self-invoice (Article 17 of Presidential Decree no. 633/72 for invoices with Natura subcodes 'N6')"),
+            ('TD17', "TD17 - Self-invoice for purchases of foreign services (both within and outside the EU) - Alternative to esterometro"),
+            ('TD18', "TD18 - Self-invoice for the purchase of intra-community goods - Alternative to esterometro"),
+            ('TD19', "TD19 - Self-invoice for foreign goods (both intra and non-EU) already present in Italy (Article 17 paragraph 2 of Presidential Decree 633/72) - Alternative to esterometro"),
+            ('TD20', "TD20 - Self-invoice Report (ex art.6 c8 471/97 or art.46 c5 331/93), or for invoice not received or for a lower amount"),
+            ('TD21', "TD21 - Self-invoice for ceiling clearance"),
+            ('TD22', "TD22 - Self-invoice for extraction of goods from VAT warehouse without VAT liability"),
+            ('TD23', "TD23 - Self-invoice for extraction of goods from VAT warehouse with VAT liability"),
+            ('TD28', "TD28 - Self-invoice for Italian tax from foreign suppliers identified but not established in Italy and for purchases from San Marino with VAT (paper invoice)")
+        ],
+        compute='_compute_l10n_it_document_type',
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends('line_ids.matching_number', 'payment_state')
+    def _compute_l10n_it_payment_method(self):
+        for move in self:
+            matching_line = move.line_ids.filtered('matching_number')
+            if matching_line and move.payment_state in {'in_payment', 'partial'}:
+                matched_ids = matching_line.matched_credit_ids or matching_line.matched_debit_ids
+                # In case of partial we take the first one to get the payment method
+                payment_move_id = matched_ids[0].credit_move_id.move_id if matched_ids[0].credit_move_id not in move.line_ids else matched_ids[0].debit_move_id.move_id
+
+                move.l10n_it_payment_method = payment_move_id.l10n_it_payment_method
+            else:
+                move.l10n_it_payment_method = move.payment_id.payment_method_line_id.l10n_it_payment_method or move.l10n_it_payment_method or 'MP05'
```

Comment:
```suggestion
        for move in self:
            matching_numbers = move.line_ids.filtered('matching_number').mapped('matching_number')
            if matching_numbers:
                matching_lines = self.env['account.move.line'].search([('matching_number', 'in', matching_numbers)])
                payment_method_line = matching_lines.payment_id.payment_method_line_id[0]
                move.l10n_it_payment_method = payment_method_line.l10n_it_payment_method
            else:
                move.l10n_it_payment_method = move.l10n_it_payment_method or 'MP05'
```
There would be an issue with arbitrarily using the first matched id since it could be a credit note.
I also think we could take advantage of the `matching_number` to find the payment method more easily :nerd_face: